jobrunner/job_executor.py

```diff
@@ -0,0 +1,170 @@
+        in which case it should be run with a network allowing access to the database and any configuration needed to
+        contact and authenticate with the database should be provided as environment variables.
+
+        The job must be run with a workspace directory at /workspace in the filesystem (this is expected to be a
+        volume mounted into the container, but other implementations are allowed). The workspace must contain a
+        checkout of the study and any inputs specified in the job definition, copied from the workspace in long-term
+        storage. If any of the specified inputs is not present in long-term storage then a JobError must be raised with
+        details of the missing file.
+
+        Any files that the job produces that match the output spec in the definition must be copied to the workspace
+        long-term storage. Anything written by the container to stdout or stderr must be captured and written to a
+        log file, metadata/{action}.log, in the workspace in long-term storage.
+
+        The action log file and any files in the output spec marked as medium privacy must also be made available in the
+        medium privacy view of the workspace in long-term storage.
+
+        The action log file and any useful metadata from the job run should also be written to a separate log storage
+        area in long-term storage.
+
+            Raises:
+                JobError: if the job definition is invalid or job creation fails
+        """
+        ...
+
+    def terminate(self, job: JobDefinition) -> None:
+        """
+        Terminate a running job.
+
+        This method must be idempotent; it may be called for a job that doesn't exist or which has already been
+        terminated in which case it must return silently.
+
+            Raises:
+                JobError: if job termination fails
+        """
+        ...
+
+    def get_status(self, job: JobDefinition) -> Tuple[State, Optional[JobResults]]:
+        """
+        Return the status of a job and the results if it has finished.
+
+        This method must be idempotent; it may be called more than once for a job even after it has finished, so any
+        irreversible cleanup which loses information about must be deferred to JobAPI.cleanup() which will only be
+        called once the results have been persisted.
+
+        The results must include a list of output files that the job produced which matched its output spec. It
+        should also include a list of files that it produced but which did not match the output spec,
+        to aid in debugging during study development.
+
+            Returns:
+                state (State): the state of the job
+                results (Optional[JobResults]): the results, if the job has finished
+
+            Raises:
+                JobError: if there is a problem retrieving the status of the job
```

Comment:
Could you explicitly add the "or the job does not exist", case?

Reply:
Yep, will do